installer/app/src/views/azure-create-app-tutorial.js.jsx

```diff
@@ -4,164 +4,261 @@ import Colors from './css/colors';
+                <button onClick={this.__handleAdvanceTutorialClick}>Walk me through it</button><br />
+                <button onClick={this.__skipTutorial}>Skip tutorial</button>
+            </div>
+        );
+
+        intro = (state.tutorialSlide || state.skipTutorial) ? null : intro;
+        tutorialSlide = (state.tutorialSlide) ? tutorialSlides[state.tutorialSlide] : null;
+
+        if (state.tutorialSlide || state.skipTutorial) {
+            pagination.prev = (state.skipTutorial && !state.tutorialSlide) ? <button id="azureTutPrev" onClick={this.__skipTutorial}>Back</button> : '';
+            pagination.prev = (state.tutorialSlide > 1) ? <button id="azureTutPrev" onClick={this.__handleGoBackTutorialClick}>Back</button> : pagination.prev;
+            pagination.next = (state.tutorialSlide < 9) ? <button id="azureTutNext" onClick={this.__handleAdvanceTutorialClick}>Next</button> : '';
+            pagination.submit = (state.skipTutorial || state.tutorialSlide === 9) ? <button type="submit" style={{margin: "0 0 0 503px"}}>Authenticate</button> : '';
+        }
+
+        return (
+            <div id={this.state.styleEl.id}>
+                {intro}
+                {tutorialSlide}
+
+                <div id="azure-tutorial-inputs">
+                    <label htmlFor="redirectURI" style={inputStyles.redirectURI}>Redirect URI</label>
+                    <input ref="redirectURI" name="redirectURI" type="text" value={redirectURI} onClick={this.__handleRedirectURIInputClick}  style={inputStyles.redirectURI} />
+                    <label htmlFor="client_id" style={inputStyles.clientId}>App Client ID</label>
+                    <input ref="client_id" name="client_id" type="text" placeholder="ab7c1052-1fe7-4642-91f6-065c94de25d4" style={inputStyles.clientId} />
+                    <label htmlFor="endpoint" style={inputStyles.endpoint}>OAuth 2.0 Token Endpoint</label>
+                    <input ref="endpoint" name="endpoint" type="text" placeholder="https://login.microsoftonline.com/{your-uid}/oauth2/token?api-version=1.0" style={inputStyles.endpoint} />
+                </div>
+
+                {pagination.prev} {pagination.next} {pagination.submit}
+            </div>
+        );
+    },
+
+    componentDidMount: function () {
+        this.state.styleEl.commit();
+    },
+
+    /**
+     * Modifies the state's `tutorialSlide` variable, advancing in the tutorial's walkthrough.
+     */
+    __handleAdvanceTutorialClick: function () {
+        var state = this.state;
+
+        console.log(this.refs);
+
+        // Validate Inputs
+        if (state.tutorialSlide === 5 && this.refs.client_id.getDOMNode().value === '') {
+            this.refs.client_id.getDOMNode().focus();
+            return;
+        } else if (state.tutorialSlide === 8 && this.refs.client_id.getDOMNode().value === '') {
+            this.refs.client_id.getDOMNode().focus();
+            return;
+        }
```

Comment:
Both branches of this if statement are the same, please combine the conditionals.